consensus/polybft/contractsapi/gen2/main.go

```diff
@@ -0,0 +1,262 @@
+package main
+
+import (
+	"bytes"
+	"fmt"
+	"go/format"
+	"io/ioutil"
+	"path/filepath"
+	"strconv"
+	"strings"
+	"text/template"
+
+	"github.com/0xPolygon/polygon-edge/consensus/polybft/contractsapi/artifact"
+	"github.com/umbracle/ethgo/abi"
+)
+
+func main() {
+	cases := []struct {
+		contract string
+		methods  []string
+		events   []string
+	}{
+		{
+			"child/StateReceiver",
+			[]string{
+				"commit",
+			},
+			[]string{},
+		},
+		{
+			"child/ChildValidatorSet",
+			[]string{
+				"commitEpoch",
+			},
+			[]string{},
+		},
+		{
+			"root/StateSender",
+			[]string{},
+			[]string{
+				"StateSynced",
+			},
+		},
+	}
+
+	rr := render{}
+
+	res := []string{}
+
+	for _, c := range cases {
+		base := filepath.Base(c.contract)
+
+		contract, err := ioutil.ReadFile("./core-contracts/artifacts/contracts/" + c.contract + ".sol/" + base + ".json")
+		if err != nil {
+			panic(err)
+		}
+
+		ar, err := artifact.DecodeArtifact(contract)
+		if err != nil {
+			panic(err)
+		}
+
+		for _, method := range c.methods {
+			res = append(res, rr.Gen(ar.Abi.Methods[method]))
+		}
+
+		for _, event := range c.events {
+			res = append(res, rr.GenEvent(ar.Abi.Events[event]))
+		}
+	}
+
+	str := `// Code generated by scapi/gen. DO NOT EDIT.
```

Comment:
does the generated file was done with this template? it seems a bit different. could we add a command to regenerate mocks to the Makefile and CI, which must fail if there is a difference between committed and generated code?

Reply:
Which parts do you see different from the template?